From 987d7db5ae466a7a7a420fbbe57cadd8951d2bfa Mon Sep 17 00:00:00 2001
From: Emmanuel Jillela <emmanuel.jillela@intel.com>
Date: Mon, 27 Jun 2016 16:53:26 -0700
Subject: [PATCH 440/639] netip_subsytem: Allow netip soft off anytime

Currently netip subsystem allows the netip soft off message only after
dualboot hand shake is done. Because of this if power manager wants to
reboot netip it cannot do so if dual boot handshake is not completed.
The change allows netip to send netip soft off message when ever power
manager wants.

Signed-off-by: Emmanuel Jillela <emmanuel.jillela@intel.com>
---
 arch/x86/NetIP_SubSystem/netip_subsystem_pm.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/x86/NetIP_SubSystem/netip_subsystem_pm.c b/arch/x86/NetIP_SubSystem/netip_subsystem_pm.c
index d15c61c..ab522f8 100644
--- a/arch/x86/NetIP_SubSystem/netip_subsystem_pm.c
+++ b/arch/x86/NetIP_SubSystem/netip_subsystem_pm.c
@@ -1090,6 +1090,8 @@ int __netss_set_service_level(uint32_t level)
 			 */
 			ret = netss_send_service_request_msg(level);
 		}
+	} else if (g_netss_sysfs_attributes_info.netip_soft_off) {
+		ret = netss_send_service_request_msg(level);
 	}
 
 	return ret;
@@ -1197,7 +1199,12 @@ int netss_runtime_suspend(struct device *dev)
       /** 2. Turn off NetIP clock */
       netss_turn_off_netip_clock();
       /** Resume will be like a fresh boot for NETIP, which involues netip properties, configure messages */
+      /** Reset boot handshake variables */
       g_netss_boot_handshake_done = false;
+      g_netip_services_available = 0;
+      g_netip_services_init_complete = 0;
+      g_netip_services_status_ready = 0;
+      g_netip_services_status_received = 0;
    }
 
    mutex_unlock(&g_netss_sysfs_attributes_info.lock);
-- 
2.10.1

